src/openms/source/FORMAT/HANDLERS/XMLHandler.cpp

```diff
@@ -262,12 +262,26 @@ namespace OpenMS
       // and all bytes except the least significant one will be zero. Thus
       // we can convert to char directly (only keeping the least
       // significant byte).
+      const XMLCh* it = chars;
+      const XMLCh* end = it + length;
+
       size_t curr_size = result.size();
       result.resize(curr_size + length);
-      for (size_t i = 0; i < length; i++)
-      {
-        result[curr_size + i] = (char)chars[i];
+      std::string::iterator str_it = result.begin();
+      std::advance(str_it, curr_size);
+      while (it!=end)
+      {   
+        *str_it = (char)*it;
+        ++str_it;
+        ++it;
       }
+
+      // This is ca. 50 % faster than 
```

Comment:
possible, could also be an aliasing issue e.g. the compiler could not exclude that `chars` and `result` point to the same array and thus had to fetch `chars[i]` each time again. 

Honestly I dont know but the speed measurements stand and I think we should not change it any more. It is equally readable if the non-iterator version that does the same thing is presented as comment